dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -762,19 +795,38 @@ ALWAYS_INLINE void Aggregator::executeImplBatch(
 
         auto & emplace_result = emplace_result_holder.value();
 
-        /// If a new key is inserted, initialize the states of the aggregate functions, and possibly something related to the key.
-        if (emplace_result.isInserted())
+        if constexpr (only_lookup)
         {
-            /// exception-safety - if you can not allocate memory or create states, then destructors will not be called.
-            emplace_result.setMapped(nullptr);
+            if (emplace_result.isFound())
+            {
+                aggregate_data = emplace_result.getMapped();
+            }
+            else
+            {
+                agg_process_info.not_found_rows.push_back(i);
+            }
+        }
+        else
+        {
+            /// If a new key is inserted, initialize the states of the aggregate functions, and possibly something related to the key.
+            if (emplace_result.isInserted())
+            {
+                /// exception-safety - if you can not allocate memory or create states, then destructors will not be called.
+                emplace_result.setMapped(nullptr);
 
-            aggregate_data = aggregates_pool->alignedAlloc(total_size_of_aggregate_states, align_aggregate_states);
-            createAggregateStates(aggregate_data);
+                aggregate_data = aggregates_pool->alignedAlloc(total_size_of_aggregate_states, align_aggregate_states);
+                createAggregateStates(aggregate_data);
 
-            emplace_result.setMapped(aggregate_data);
+                emplace_result.setMapped(aggregate_data);
+            }
+            else
+            {
+                aggregate_data = emplace_result.getMapped();
+
+                if constexpr (collect_hit_rate)
+                    ++agg_process_info.hit_row_cnt;
+            }
```

Comment:
inst->batch_that->addBatch in L841 don't need to be aware of the not_found_rows ?

Reply:
I think no need to be aware of `not_found_rows`. Because for these rows, the AggregateDataPtr will be nullptr, and in `IAggregateFunctionHelper:addBatch`, it will ignore nullptr AggregateDataPtr.